internal/pkg/agent/install/flavors.go

```diff
@@ -0,0 +1,258 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package install
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	v1 "github.com/elastic/elastic-agent/pkg/api/v1"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	FlavorBasic   = "basic"
+	FlavorServers = "servers"
+
+	DefaultFlavor  = FlavorBasic
+	flavorFileName = ".flavor"
+)
+
+type SkipFn func(relPath string) bool
+
+var ErrUnknownFlavor = fmt.Errorf("unknown flavor")
+
+type FlavorDefinition struct {
+	Name       string
+	Components []string
+}
+
+func UsedFlavor(topPath, defaultFlavor string) (string, error) {
+	filename := filepath.Join(topPath, flavorFileName)
+	content, err := os.ReadFile(filename)
+	if err != nil {
+		// file does not exist, flavor was not marked probably due to earlier version
+		// fallback to default if defined
+		if defaultFlavor != "" && os.IsNotExist(err) {
+			return defaultFlavor, nil
+		}
+
+		// failed reading flavor, do not break behavior and apply none as widest
+		return "", err
+	}
+
+	return string(content), nil
+}
+
+func Flavor(detectedFlavor string, registryPath string, flavorsRegistry map[string][]string) (FlavorDefinition, error) {
+	if flavorsRegistry == nil {
+		f, err := os.Open(registryPath)
+		if err != nil {
+			return FlavorDefinition{}, err
+		}
+		manifest, err := v1.ParseManifest(f)
+		if err != nil {
+			return FlavorDefinition{}, err
+		}
+		defer f.Close()
+		flavorsRegistry = manifest.Package.Flavors
+	}
+
+	components, found := flavorsRegistry[detectedFlavor]
+	if !found {
+		return FlavorDefinition{}, ErrUnknownFlavor
+	}
+
+	return FlavorDefinition{detectedFlavor, components}, nil
+}
+
+// SpecsForFlavor returns spec files associated with specific flavor
+func SpecsForFlavor(flavor FlavorDefinition) ([]string, error) {
```

Comment:
This function always returns `nil` as error, is the error needed?

Reply:
leftover from previous design, thanks for pointing it out